### DISCUSSION TITLE
Database Credentials

### DISCUSSION THREAD

**Discussion by freddieb77** [mdwiltfong/secret-santa#15] [Q&A] [answered]
Maybe I missed this on Wednesday. I am not seeing where or what the credentials are for the Database. On Step three when I run the `npm run migrate:dev` command. I get a P1000 Error for Authentication failed against database server. Any help to point me in the right direction would be appreciated.

Freddie

**Comment by mdwiltfong** [comment, marked as answer]
Hey @freddieb77 ! The app is trying to connect to your local database using the connection string in the `.env` file located at the root of the project. So after you create the `.env` at the root, you will have to provide the correct connection string. The connection string should look like [this](https://github.com/mdwiltfong/secret-santa/blob/025e4dba622e98c62fc896eeeb3341496cba23ed/.env.example#L7). As for what are the credentials, i can only guess. There is typically a default `postgres` user in psql. So you can try to connect using `postgres` as the username. If you don't know the password then maybe you need to change it. Here are some steps on how to do that: 

```
Resetting Password
If you cannot find the existing credentials or if you need to reset the password for some reason, you can do the following:

Open a Command Line Interface:

On Windows: Open psql shell or Windows Command Prompt.
On Linux/Mac: Open a terminal.
Log in to PostgreSQL:

Use the command sudo -u postgres psql. This command logs you into PostgreSQL as the 'postgres' user.
Reset Password:

Inside the psql shell, type: ALTER USER postgres PASSWORD 'newpassword';
Replace newpassword with your desired password.
Restart PostgreSQL Service (if needed):

On Windows: Restart the service from the services management tool.
On Linux/Mac: You might need to use a command like sudo service postgresql restart.
```

**Reply by freddieb77** [reply]
Ok, Thanks for the response. I found what the default password and user should be. I had just installed psql, so it should be the default setup. I must have something wrong in my connection string in the .env.  I will try that again and if it does not work then I will try resetting it. Thanks again for the help.

**Comment by gracekishino** [comment]
Just sharing this in case it helps @freddieb77
 
 I didn't need a password for  my connection string (`DATABASE_URL="postgres://grace@localhost/secret-santa?schema=public"`), I think I originally set up my postgres user (`grace`) with no password requirement..
 
 @mdwiltfong hope this is OK as it is just a dev environment locally on my laptop, or do you think it better that I change the setup to require a password?
 
   I created a separate DB called `secret-santa` for this project so that's why it's in my connection string.

**Reply by mdwiltfong** [reply]
Ah yeah that's another possibility. I personally like adding a password, because that way if I'm facing some sort of connection issue, it's not because I'm passing on the credentials incorrectly. 

I'll add this to the documentation!

**Reply by gracekishino** [reply]
Thanks that makes sense! I'll try to set it up with a password 😊

**Reply by freddieb77** [reply]
Micheal and Grace, 
Thank you for the tips. After I looked through the Postgresql documentation some more and realized that on Linux when you install it for the first time it has a default user of postgres and it does not create a password. So I  did use the commands you posted and set a password. Then updated the connection string in my .env and everything is perfect. Thanks again.

Freddie

**Reply by mdwiltfong** [reply]
Excellent! I'll definitely add that case to the docs!
